goro/src/androidTest/java/com/stanfy/enroscar/goro/OnDemandBindingGoroAndroidTest.java

```diff
@@ -0,0 +1,78 @@
+package com.stanfy.enroscar.goro;
+
+import android.annotation.TargetApi;
+import android.app.Application;
+import android.os.AsyncTask;
+import android.os.Build;
+import android.test.ApplicationTestCase;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+public class OnDemandBindingGoroAndroidTest extends ApplicationTestCase<Application> {
+
+  private OnDemandBindingGoro goro;
+
+  public OnDemandBindingGoroAndroidTest() {
+    super(Application.class);
+  }
+
+  @Override
+  protected void setUp() throws Exception {
+    super.setUp();
+    createApplication();
+    GoroService.setDelegateExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+    this.goro = (OnDemandBindingGoro) Goro.bindOnDemandWith(getApplication());
+  }
+
+  @Override
+  protected void tearDown() throws Exception {
+    super.tearDown();
+  }
+
+  public void testSchedule() throws InterruptedException {
+    final CountDownLatch sync = new CountDownLatch(1);
+    Callable<String> task = new Callable<String>() {
+      @Override
+      public String call() {
+        return "done";
+      }
+    };
+    final Object[] result = new Object[1];
+    goro.schedule(task)
+        .subscribe(new FutureObserver<String>() {
```

Comment:
Seems like you can use `rx.observers.TestSubscriber` here. It has `awaitTerminalEvent(long timeout, TimeUnit unit)` method.